front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts

```diff
@@ -0,0 +1,130 @@
+  const upperIndex = (tenIndex + 1) * 10 < trainIds.length ? (tenIndex + 1) * 10 : undefined;
+  return trainIds.slice(tenIndex * 10, upperIndex);
+};
+
+type useLazyLoadTrainsProp = {
+  infraId?: number;
+  trainIdsToFetch?: number[];
+  path?: PathfindingResultSuccess;
+  trainSchedules?: TrainScheduleResult[];
+  setTrainIdsToFetch: Dispatch<SetStateAction<number[] | undefined>>;
+};
+
+/**
+ * This hook gradually fetches and projects trains of the timetable.
+ *
+ * It first fetches the simulation of 10 trains at a time, then projects them on the path.
+ * This optimizes the performance of the application and allow us to display the trains as
+ * soon as they are ready.
+ */
+const useLazyLoadTrains = ({
+  infraId,
+  trainIdsToFetch,
+  path,
+  trainSchedules,
+  setTrainIdsToFetch,
+}: useLazyLoadTrainsProp) => {
+  const { getElectricalProfileSetId } = useOsrdConfSelectors();
+  const electricalProfileSetId = useSelector(getElectricalProfileSetId);
+
+  const [trainScheduleSummariesById, setTrainScheduleSummariesById] = useState<
+    Dictionary<TrainScheduleWithDetails>
+  >({});
+  const [trainIdsToProject, setTrainIdsToProject] = useState<number[]>([]);
+  const [allTrainsLoaded, setAllTrainsLoaded] = useState(false);
+
+  const trainScheduleSummaries = useMemo(
+    () => sortBy(Object.values(trainScheduleSummariesById), 'startTime'),
+    [trainScheduleSummariesById]
+  );
+
+  const [postV2TrainScheduleSimulationSummary] =
+    osrdEditoastApi.endpoints.postV2TrainScheduleSimulationSummary.useLazyQuery();
+
+  const { data: { results: rollingStocks } = { results: [] } } =
+    osrdEditoastApi.endpoints.getLightRollingStock.useQuery({ pageSize: 1000 });
+
+  const trainSchedulesById = useMemo(() => keyBy(trainSchedules, 'id'), [trainSchedules]);
+
+  const { projectedTrains, setProjectedTrainsById } = useLazyProjectTrains({
+    infraId,
+    trainIdsToProject,
+    path,
+    trainSchedules,
+    moreTrainsToCome: !allTrainsLoaded,
+    setTrainIdsToProject,
+  });
+
+  // gradually fetch the simulation of the trains
+  useEffect(() => {
+    const getTrainScheduleSummaries = async (_infraId: number, _trainToFetchIds: number[]) => {
+      let formattedSummaries: Dictionary<TrainScheduleWithDetails> = {
+        ...trainScheduleSummariesById,
+      };
+      setAllTrainsLoaded(false);
+
+      for (let i = 0; i * 10 <= _trainToFetchIds.length - 1; i += 1) {
+        const packageToFetch = getTrainIdsPackage(i, _trainToFetchIds);
+
+        const test = await postV2TrainScheduleSimulationSummary({
+          body: {
+            infra_id: _infraId,
+            ids: packageToFetch,
+            electrical_profile_set_id: electricalProfileSetId,
+          },
+        }).unwrap();
+
+        // launch the projection of the trains
+        setTrainIdsToProject((prev) => [...prev, ...packageToFetch]);
+        // console.log('1. package fetched', packageToFetch);
```

Comment:
```suggestion
```

Reply:
This needs to be addressed before merging as well.

Reply:
Done :white_check_mark: